File: src/models/socialProfile.ts
export type SocialPlatform = 'facebook' | 'instagram' | 'youtube' | 'telegram' | 'x' | 'tiktok' | 'web';

export default interface SocialProfile {
  id: string;
  name: string;
  coverPhotoUrl: string;
  avatarUrl: string;
  bio?: string;
  followersCount: number;
  followingCount: number;
  garageCount: number;
  ridesCount: number;
  garageDistance?: string;
  garageDuration?: string;
  garageElevation?: string;
  rideDistance?: string;
  rideDuration?: string;
  rideElevation?: string;
  socialLinks: Array<{ platform: SocialPlatform; url: string }>;
}

File: src/mvc/Entities/SocialProfile.cs
using System.ComponentModel.DataAnnotations;

namespace BikerHub.Entities;

public class SocialProfile
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(512)]
    public string CoverPhotoUrl { get; set; } = string.Empty;

    [MaxLength(512)]
    public string AvatarUrl { get; set; } = string.Empty;

    public string? Bio { get; set; }
    public int FollowersCount { get; set; }
    public int FollowingCount { get; set; }
    public int GarageCount { get; set; }
    public int RidesCount { get; set; }
    public string? GarageDistance { get; set; }
    public string? GarageDuration { get; set; }
    public string? GarageElevation { get; set; }
    public string? RideDistance { get; set; }
    public string? RideDuration { get; set; }
    public string? RideElevation { get; set; }
    public string SocialLinksJson { get; set; } = "[]";
}
